### ISSUE TITLE
App freezing while in background on Galaxy S22

### ISSUE THREAD

**Issue by RyanThaDude** [bluelinepc/signalcheck#58]
**Describe the bug**
When running the app for a period of time in the background it will freeze to the point that it requires a force close. This seems to happen with a larger database, not so much with an empty. The app will stop logging when it freezes. I am running two SIMs. 

Also, no other issues on any other my phones, just the S22. The phone also must be moving between sites and not stationary. I do edit the database but ruled out corruption since using a clean database that grows will eventually do the same thing. 

**To Reproduce**
With a very large database run the app in the background for a period of time and try to bring it back to the foreground. 

**Expected behavior**
The app should return to the foreground without freezing. 

**Screenshots**
N/A

**Device details**
Diag sent from app

**Additional context**
Please add any other context about the problem here.

**Comment by mikejeep** [comment]
Can you try disabling your secondary SIM and see if the behavior changes at all?

**Comment by RyanThaDude** [comment]
I will give this a try this weekend at some point while out and about.

**Comment by RyanThaDude** [comment]
My trip today wasn't as long as I had hoped. I didn't notice any freezing when the 2nd SIM was disabled, but again, this was a short trip. I did notice that the app was still somehow logging the second SIM, but I'm throwing this as stale data as it was only one site logged, but it is the only site in the area I traveled. None of the other apps I had running (Cellmapper, Netmonster) did not log any data from the inactive SIM.

**Comment by RyanThaDude** [comment]
Adding on to this bug. Sadly, switching off the 2nd SIM didn't stop the app from freezing. I didn't notice when it exactly froze but it seems like when there's frequent changes in sites. I had my wife with me and I didn't exactly want to scare her while checking the phone while behind the wheel 🤣

**Comment by mikejeep** [comment]
Circling back on this -- are you still experiencing this issue? Are you using the latest beta release, or are you on the non-beta version? Thanks.

**Comment by RyanThaDude** [comment]
Yes, it's still an issue, even with the latest OneUI update. I'm not part of the beta. (4.841 May 13)

**Comment by mikejeep** [comment]
Could you please try updating to the latest version (4.85), released on October 12, and confirm if it has been fixed or if the issue persists? Thanks!

**Comment by RyanThaDude** [comment]
I had it updated on day 1, but didn't get a real good chance to test it until today. The tests keeping the app in the background for a short time, like a few seconds seemed fine, but sadly, it still freezes on extended times while using both my pSIM (Verizon) and eSIM (T-Mobile MVNO) and just my pSIM. I had hope to do more testing and mapping today but my vehicle decided it had other plans for me. :-\
